core/src/main/scala/za/co/absa/spline/harvester/logging/ObjectStructureLogging.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.harvester.logging
+
+import org.apache.spark.internal.Logging
+
+import scala.util.control.NonFatal
+import scala.util.{Failure, Success, Try}
+
+trait ObjectStructureLogging {
+  self: Logging =>
+
+  protected def logTraceObjectStructure(obj: => AnyRef): Unit =
+    if (log.isTraceEnabled) logObjectStructure(obj, log.trace, log.trace)
```

Comment:
Though `log` has a type of `slf4j.Logger` the trait itself is combined with `spark.Logging` that has `logXXX()` methods doing the same thing. So we can use them directly instead of the `log` (that is one level more distant dependency).
```suggestion
    logObjectStructure(obj, logTrace, logTrace)
```